drivers/led/ncp5623.c

```diff
@@ -149,9 +149,9 @@ static int ncp5623_led_init(const struct device *dev)
 			return -ENODEV;
 		}
 
-		if (led_info->num_colors != 3 || led_info->num_colors != 1) {
-			LOG_ERR("%s: invalid number of colors %d (must be %d or 1)", dev->name,
-				led_info->num_colors, NCP5623_CHANNEL_COUNT);
+		if (led_info->num_colors != NCP5623_CHANNEL_COUNT) {
+			LOG_ERR("%s: invalid number of colors %d (must be %d with a single LED)",
+				dev->name, led_info->num_colors, NCP5623_CHANNEL_COUNT);
```

Comment:
Please can you add the missing return as well ?

Reply:
done